### DISCUSSION TITLE
execution reverted while calling balanceOf a ERC20 Token using go-ethereum

### DISCUSSION THREAD

**Discussion by Iamshashvat** [zkSync-Community-Hub/zksync-developers#553] [SDKs] [answered]
I'm using the same script to call balanceOf for ERC20 tokens on other chains, and I'm successfully getting the data. However, when I try to do this for zkSync, I'm getting an "execution reverted" error. It's surprising because write operations are working fine. I'm using github.com/ethereum/go-ethereum v1.10.26.

**Comment by bxpana** [comment]
Thanks for the post. Will check this out and get back to you 😁

**Reply by bxpana** [reply, marked as answer]
You'll want to make sure you're using our SDK and then it should work as expected
